source/collection-management/tests/timeseries-collection.yml

```diff
@@ -34,10 +34,11 @@ tests:
         object: *database0
         arguments:
           collection: *collection0Name
+          expireAfterSeconds:
+            $numberLong: "604800"
```

Comment:
Can you confirm whether or not the server actually requires this to be an Int64 type (i.e. 0x12 instead of 0x10 for Int32)? I would expect the server to accept either integer type, in which case I don't think there is any need to use `$numberLong` here.

Keep in mind that drivers are likely to parse this Extended JSON into their native integer type. That may have no direct impact on the type used for the outgoing command, particular if drivers decide to encode integers into Int32 or Int64 BSON types based on the integer's value.

If you're concerned with asserting the outgoing type, you should consider using `$numberLong` in the APM assertion to match the outgoing command.

----

If the intention was the document the type until the actual spec is written, I think that can be accomplished with a comment instead.

I'm actually not sure if the server would even end up documenting the particular type, as the similarly named option for [`createIndexes`](https://docs.mongodb.com/v5.0/reference/command/createIndexes/) is only reported as an "integer". That's not to say we shouldn't do so ourselves (e.g. CRUD mentions when a value can be 64-bit).

Reply:
For context, this switch to `$numberLong` came out of a [discussion](https://github.com/mongodb/mongo-go-driver/pull/675#discussion_r640190211) in the Go driver implementation.

The server will accept either integer type, but it does [store](https://github.com/mongodb/mongo/blob/1a6e08a61c138e7da0500c11de78d3489b7024d4/src/mongo/db/catalog/collection_options.h#L156) the type in collection_options as an int64. Since it does accept both types, I won't use `$numberLong` here, but I did add a comment saying that the type should be int64 and used `$numberLong` in the APM assertion for the outgoing command for the sake of consistency across drivers

Reply:
In light of [Flexible Numeric Comparisons](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#flexible-numeric-comparisons), I actually think it's pointless to use `$numberLong` to assert the outgoing type. It's rare that we do this for spec tests, although I do see things like `{$numberLong: "1" }` used for `lsid` in some of our legacy transaction spec tests. IMO, it makes no sense to do this unless we actually need to represent a large number in Extended JSON.

If we actually wanted to match a particular BSON type, we have [`$$type`](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#type), but that can't currently be combined with a value check. It's essentially a wildcard (like `$$exists: true`) with a type requirement.

If it helps to have prior art for this, consider all of the instances of "Int64" for various options in the [CRUD](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst) spec. We don't ever assert those types in our spec tests, nor do we have tests that use values larger than `INT32_MAX`. It's purely documentation for the driver's implementation.

Since there's not yet a spec here, I'd suggest just having a comment and avoiding `$numberLong` altogether.



I gave some more thought to using `$numberLong` for the APM assertion (because we very seldom do that in other spec tests) because we don't do that at all for